### PR TITLE
Fix `.extract()` for query strings containing `?`

### PR DESCRIPTION
**Pull request by neykov**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -114,7 +114,13 @@ function keysSorter(input) {
...
 
 exports.extract = function (str) {
-	return str.split('?')[1] || '';
+	var queryStart = str.indexOf('?');
+	if (queryStart == -1) {
+		// no query string
+		return '';
+	} else {
+		return str.substring(queryStart+1)
+	}
```
Comment (neykov) [inline index.js]: Even more logic would be needed if the library expects to work with URLs containing hashes (`#`).

File: index.js [hunk 2]
```diff
@@ -114,7 +114,13 @@ function keysSorter(input) {
 }
 
 exports.extract = function (str) {
-	return str.split('?')[1] || '';
+	var queryStart = str.indexOf('?');
+	if (queryStart == -1) {
+		// no query string
+		return '';
+	} else {
+		return str.substring(queryStart+1)
```
Comment (sindresorhus) [inline index.js]: Can you use `.slice()` instead here. Just my preference.

File: index.js [hunk 3]
```diff
@@ -114,7 +114,13 @@ function keysSorter(input) {
 }
 
 exports.extract = function (str) {
-	return str.split('?')[1] || '';
+	var queryStart = str.indexOf('?');
+	if (queryStart == -1) {
+		// no query string
```
Comment (sindresorhus) [inline index.js]: Moot comment. The code is already clear enough.

File: index.js [hunk 4]
```diff
@@ -114,7 +114,13 @@ function keysSorter(input) {
 }
 
 exports.extract = function (str) {
-	return str.split('?')[1] || '';
+	var queryStart = str.indexOf('?');
+	if (queryStart == -1) {
```
Comment (SamVerschueren) [inline index.js]: Use `===`

File: index.js [hunk 5]
```diff
@@ -114,7 +114,13 @@ function keysSorter(input) {
...
+		// no query string
+		return '';
+	} else {
```
Comment (SamVerschueren) [inline index.js]: Do not use the `else`

```
if (queryStart === -1) {
    return '';
}

return str.slice(queryStart+1);
```

**Comment by neykov** [comment]
Addressed comments.

File: index.js [hunk 6]
```diff
@@ -114,7 +114,11 @@ function keysSorter(input) {
 }
 
 exports.extract = function (str) {
-	return str.split('?')[1] || '';
+	var queryStart = str.indexOf('?');
+	if (queryStart === -1) {
+		return '';
+	}
+	return str.slice(queryStart+1);
```
Comment (kevva) [inline index.js]: Could do `return queryStart === -1 ? '' : str.slice(queryStart + 1);`

**Comment by apuntovanini** [comment]
Any plans to merge this? Thanks!

**Comment by rapzo** [comment]
I would if i could :-)

**Comment by jorrit** [comment]
Wouldn't `return str.split('?', 2)[1] || '';` have sufficed?

**Pull request merged**